packages/core-mobile/app/store/rpc/listeners/request/handleRequestViaVMModule.ts

```diff
@@ -90,3 +102,33 @@ export const handleRequestViaVMModule = async ({
     })
   }
 }
+
+const getContext = (
+  method: VmModuleRpcMethod,
+  params: unknown,
+  activeAccount: CorePrimaryAccount | undefined
+): Record<string, string> | undefined => {
+  if (method === VmModuleRpcMethod.AVALANCHE_SEND_TRANSACTION) {
+    if (
+      !params ||
+      typeof params !== 'object' ||
+      'chainAlias' in params === false
+    ) {
+      return undefined
+    }
+
+    const vm = Avalanche.getVmByChainAlias(params.chainAlias as string)
+    const currentAddress = getAddressByVM(vm, activeAccount)
+
+    if (!currentAddress) {
+      return undefined
+    }
+
+    return {
+      currentAddress,
+      xpubXP: MnemonicWalletInstance.xpubXP
```

Comment:
the xpubXP might be sensitive stuff that we don't want to expose to vm module. also context is more for metadata or UI related data.

Reply:
if we could figure out a better way, that would be preferred

Reply:
The get addressesByIndices is actually gonna be deprecated soon. You can actually just use getAddresses function in this avalanche module.